pennylane/beta/tapes/qnode.py

```diff
@@ -383,11 +383,38 @@ def construct(self, args, kwargs):
             )
 
     def __call__(self, *args, **kwargs):
+
+        if self.interface == "autograd":
+            # HOTFIX: to maintain compatibility with core, here we treat
+            # all inputs that do not explicitly specify `requires_grad=False`
+            # as trainable. This should be removed at some point, forcing users
+            # to specify `requires_grad=True` for trainable parameters.
+            args = [
+                anp.array(a, requires_grad=True) if not hasattr(a, "requires_grad") else a
+                for a in args
+            ]
+
         # construct the tape
         self.construct(args, kwargs)
 
         # execute the tape
-        return self.qtape.execute(device=self.device)
+        res = self.qtape.execute(device=self.device)
+
+        if list(res.shape) == [1]:
+            # HOTFIX: to maintain compatibility with core, we squeeze

```

Comment:
Same here

Reply:
Do I understand correctly, the new qnode would find it more natural to have unsqueezed outputs, but the old qnode uses squeezed ones? Which option is better? Which one do we want to support in the long term?

Reply:
Personally, I find it cleaner if the QNode always returned an array. A single measurement simply returns an array of length 1.

_However_ we have two many tutorials/demos that use the fact that a single measurement returns a scalar, so I am hesitant to make a change to this behaviour as part of the refactor. The refactor should ideally leave the UI and API unchanged, and just modify the internals.

But we should re-visit this and decide if we want to change the behaviour. We could do it in stages; make the squeezing optional, and being to show a deprecation warning.